src/pipelinedefinition.cpp

```diff
@@ -85,14 +87,66 @@ Status PipelineDefinition::validate(ModelManager& manager) {
     return validationResult;
 }
 
+Status PipelineDefinition::initializeNodeResources() {
+    for (const auto& nodeInfo : nodeInfos) {
+        if (nodeInfo.kind == NodeKind::CUSTOM) {
+            void* customNodeLibraryInternalManager = nullptr;
+            auto it = nodeResources.find(nodeInfo.nodeName);
+            bool internalManagerExists = (it != nodeResources.end());
+            if (internalManagerExists) {
+                customNodeLibraryInternalManager = nodeResources.at(nodeInfo.nodeName);
```

Comment:
Here you search for node name inside nodeResources twice. Once in line 94 and second time in line 97. You can just use it->second if internal manager exists.